### PR TITLE
fix: TypeError cannot read property 'focus' of undefined

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

**Issue**
The error `Cannot read property 'focus' of undefined`occurs because the `this.$refs.searchbar` is undefined at the time `focusSearchInput()` method is called. This can happen if the method is called before the component has fully mounted or if the `ref="searchbar"` is not correctly attached to any element. 

I tried multiple ways to reproduce this issue, but none worked.

**Solution**
To resolve this issue, I can add a check to ensure that `this.$refs.searchbar` is defined before calling `focus()` method.

Fixes https://linear.app/chatwoot/issue/CW-3405/typeerror-cannot-read-property-focus-of-undefined

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/shared/components/emoji/EmojiInput.vue [hunk 1]
```diff
@@ -174,7 +174,7 @@ export default {
       return categoryItem ? categoryItem.emojis[0].emoji : '';
     },
     focusSearchInput() {
-      this.$refs.searchbar.focus();
+      this.$refs.searchbar?.focus();
```
Comment (scmmishra) [inline app/javascript/shared/components/emoji/EmojiInput.vue:177]: Can we use `autofocus` property on the input instead?

Comment (iamsivin) [reply app/javascript/shared/components/emoji/EmojiInput.vue:177]: @scmmishra I tried adding the `autofocus` property for input, but it doesn't work.

Comment (scmmishra) [reply app/javascript/shared/components/emoji/EmojiInput.vue:177]: Were you able to reproduce it?. Try adding a `$nextTick` see if that fixes the issue.

Comment (iamsivin) [reply app/javascript/shared/components/emoji/EmojiInput.vue:177]: @scmmishra No, I couldn't able to reproduce it.

Comment (iamsivin) [reply app/javascript/shared/components/emoji/EmojiInput.vue]: @scmmishra Updated the PR

Comment (scmmishra) [reply app/javascript/shared/components/emoji/EmojiInput.vue]: Sounds good, if this occurs again, we will now. I will update the sentry rule

**Review by scmmishra — APPROVED** [review]

**Pull request merged**